### PR TITLE
[tune] refactor and add examples

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

Make sure examples show a complete workflow and add missing examples to page.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by amogkam — APPROVED** [review]
LGTM

File: python/ray/tune/examples/ax_example.py [hunk 1]
```diff
@@ -71,15 +68,17 @@ def easy_objective(config):
...
+    scheduler = AsyncHyperBandScheduler()
+    analysis = tune.run(
         easy_objective,
         name="ax",
+        metric="hartmann6",  # provided in the 'easy_objective' function
+        mode="min",
         search_alg=algo,
         scheduler=scheduler,
         **tune_kwargs)
+
```
Comment (krfricke) [inline python/ray/tune/examples/ax_example.py:83]: I'm not a huge fan of the `**tune_kwargs` thing. Can we just use a config dict and pass the other kwargs directly as arguments?

File: python/ray/tune/examples/cifar10_pytorch.py [hunk 2]
```diff
@@ -187,21 +186,18 @@ def main(num_samples=10, max_num_epochs=10, gpus_per_trial=2):
...
+        tune.with_parameters(train_cifar, data_dir=data_dir),
         resources_per_trial={"cpu": 2, "gpu": gpus_per_trial},
         config=config,
+        metric="loss",
+        mode="min",
         num_samples=num_samples,
-        scheduler=scheduler,
-        progress_reporter=reporter)
+        scheduler=scheduler
+    )
```
Comment (krfricke) [inline python/ray/tune/examples/cifar10_pytorch.py:200]:
```suggestion
        scheduler=scheduler)
```
inb4 linter

File: python/ray/tune/examples/skopt_example.py [hunk 3]
```diff
@@ -34,18 +33,9 @@ def easy_objective(config):
...
-    tune_kwargs = {
-        "num_samples": 10 if args.smoke_test else 50,
-        "config": {
-            "steps": 100,
-            "width": tune.uniform(0, 20),
-            "height": tune.uniform(-100, 100),
-            "activation": tune.choice(["relu", "tanh"])
-        }
-    }
+    tune_kwargs = {}
```
Comment (krfricke) [inline python/ray/tune/examples/skopt_example.py:38]: remove?

**Review by krfricke — DISMISSED** [review]
Let's get rid of the `**tune_kwargs` in a handful of these examples. Otherwise great and much needed!

File: python/ray/tune/examples/cifar10_pytorch.py (hunk 2 above)
Comment (richardliaw) [reply python/ray/tune/examples/cifar10_pytorch.py:200]: unfortunately the linter is keeping it as is

File: python/ray/tune/examples/ax_example.py (hunk 1 above)
Comment (richardliaw) [reply python/ray/tune/examples/ax_example.py:78]: yep!

File: python/ray/tune/examples/cifar10_pytorch.py (hunk 2 above)
Comment (krfricke) [reply python/ray/tune/examples/cifar10_pytorch.py:200]: My bad!

**Pull request merged**